libcodechecker/server/api/report_server.py

```diff
@@ -53,6 +53,51 @@ class CountFilter:
     RUN_HISTORY_TAG = 6
 
 
+class DBSession():
+
+    def __init__(self, session_maker):
+        self.__session = None
+        self.__session_maker = session_maker
+
+    def __enter__(self):
+        # create new session
+        self.__session = self.__session_maker()
+        return self.__session
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.__session:
+            self.__session.close()
+
+
+def ExceptionConverter(func):
+    """
+    Convert internal exceptions to RequestFailed exception
+    which can be sent back on the thrift connections.
+    """
+    func_name = func.__name__
```

Comment:
We are using multiple decorators at multiple functions:
E.g.:
```py
@ExceptionConverter
@timeit
def getRunData(self, run_filter):
   ...
```
In this case python will chain the function decorator: ExceptionConverter(timeit(getRunData()))
This way the `func_name` will be `wrapper` which is not what I expected.
```bash
[ERROR] [2017-12-14 11:01:37] {RUN ACCESS HANDLER} - report_server.py:92 wrapper() - wrapper:
```

I have checked the `timeit` decorator and I found a bug. It always uses `release_wrapper` (in case of `--verbose debug` too). Please replace the `timeit` decorator (https://github.com/Ericsson/codechecker/blob/116d1e6c95f54c328da494f2ed7aceef2f5f4d08/libcodechecker/profiler.py#L64-L67) with this:
```py
    @wraps(function)
    def wrapper(*args, **kwargs):
        if LoggerFactory.get_log_level() == logger.DEBUG:
            return debug_wrapper(*args, **kwargs)
        else:
            return release_wrapper(*args, **kwargs)

    return wrapper
```
Also use the `wraps` decorator. This way the function name will be good:
```bash
[ERROR] [2017-12-14 11:11:30] {RUN ACCESS HANDLER} - report_server.py:92 wrapper() - getRunData:
```

Reply:
I've created a separate pull request for fixing this #1260